mono/metadata/appdomain.c

```diff
@@ -530,7 +530,8 @@ mono_domain_create_appdomain_checked (char *friendly_name, char *configuration_f
 	}
 	MONO_HANDLE_SET (setup, configuration_file, config_file);
 
-	MonoAppDomainHandle ad = mono_domain_create_appdomain_internal (friendly_name, setup, error);
+	MonoAppDomainHandle ad;
+	ad = mono_domain_create_appdomain_internal (friendly_name, setup, error);
 	goto_if_nok (error, leave);
```

Comment:
Whats the difference between these two lines ?

Reply:
Cannot goto around initialization in C++. First form is illegal in C++, because there is a goto that bypasses it, leaving the variable uninitialized (hypothetically — the code is just illegal and will not compile). It is uninitialized in the second case, and in the first case for C, but C++ requires the more explicit lack of initialization.

I.e. the style is dubious but valid in C. I.e. what if you use the variables at the goto target? The code suggests they are initialized but they are not. C++ disallows it.

Reply:
I mean there is no goto between the two lines.

Reply:
Maybe I edited an extra line, but in general there is a goto from before to after.

Reply:
Alternate solution is move locals to the top, perhaps initialized with zeros, or not -- C89 style.
Or goto start of function instead of end, as I did once or twice (once in this PR, once in another I think).
All three work.
Eventually we can use "early return" i.e. with  destructors, but not yet.

These are all good solutions.

Reply:
Moving the handles to top in particular  is good, so they can have  constructors -- which will otherwise bring back the same error. Currently they are uninitialized and that is allowed..